internal/tools/aws/rds.go

```diff
@@ -0,0 +1,193 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+func (a *Client) rdsGetDBSecurityGroupIDs(logger log.FieldLogger) ([]string, error) {
+	svc := ec2.New(session.New(), &aws.Config{
+		Region: aws.String(DefaultAWSRegion),
+	})
+
+	input := &ec2.DescribeSecurityGroupsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: aws.String(DefaultDBSecurityGroupTagKey),
+				Values: []*string{
+					aws.String(DefaultDBSecurityGroupTagValue),
+				},
+			},
+		},
+	}
+
+	result, err := svc.DescribeSecurityGroups(input)
+	if err != nil {
+		return []string{}, err
+	}
+
+	var dbSecurityGroups []string
+	for _, sg := range result.SecurityGroups {
+		dbSecurityGroups = append(dbSecurityGroups, *sg.GroupId)
+	}
+
+	if len(dbSecurityGroups) == 0 {
+		return []string{}, fmt.Errorf("unable to find security groups tagged for Mattermost DB usage: %s=%s", DefaultDBSecurityGroupTagKey, DefaultDBSecurityGroupTagValue)
+	}
+
+	logger.WithField("security-group-ids", dbSecurityGroups).Debugf("Found %d DB tagged security groups", len(dbSecurityGroups))
+
+	return dbSecurityGroups, nil
+}
+
+func (a *Client) rdsEnsureDBClusterCreated(awsID, username, password string, logger log.FieldLogger) error {
+	svc := rds.New(session.New(), &aws.Config{
+		Region: aws.String(DefaultAWSRegion),
+	})
+
+	_, err := svc.DescribeDBClusters(&rds.DescribeDBClustersInput{
+		DBClusterIdentifier: aws.String(awsID),
+	})
+	if err == nil {
+		logger.WithField("db-cluster-name", awsID).Debug("AWS DB cluster already created")
+
+		return nil
+	}
+
+	dbSecurityGroupIDs, err := a.rdsGetDBSecurityGroupIDs(logger)
+	if err != nil {
+		return err
+	}
+
+	input := &rds.CreateDBClusterInput{
+		AvailabilityZones: []*string{
+			aws.String("us-east-1a"),
+			aws.String("us-east-1b"),
+			aws.String("us-east-1c"),
+		},
+		BackupRetentionPeriod: aws.Int64(7),
+		DBClusterIdentifier:   aws.String(awsID),
+		DatabaseName:          aws.String("mattermost"),
```

Comment:
Nit: is there any reason we might not want to hardcode these values?

Reply:
Yep, we are very-likely to want to make some of these values configurable in the future. Due to this already being a large change I opted to hardcode things that made sense. When we require more options in the future we will definitely add them.

Reply:
Yup that's sort of what I suspected :+1: